tcms/rpc/tests/test_testexecutionstatus.py

```diff
@@ -0,0 +1,38 @@
+# -*- coding: utf-8 -*-
+# pylint: disable=attribute-defined-outside-init, invalid-name, objects-update-used
+
+from xmlrpc.client import ProtocolError
+from tcms.testruns.models import TestExecutionStatus
+from tcms.rpc.tests.utils import APITestCase, APIPermissionsTestCase
+
+
+class TestFilter(APITestCase):
+    """Test TestExecutionStatus.filter"""
+
+    def _fixture_setup(self):
+        super()._fixture_setup()
+
+    def test_filter_statuses(self):
+        execution_status_weights = []
+        for execution_status in TestExecutionStatus.objects.all():
+            if (execution_status.weight < 0):
+                execution_status_weights.append(execution_status.weight)
+
+        self.assertTrue(len(execution_status_weights) > 0)
```

Comment:
This scenario is now collecting all negative statuses from the DB and asserting that their count is greater than zero. This is the same as counting how many negative statuses are in the database to begin with.  This is not using the TestExecutionStatus.filter API and isn't suitable for a test.

Reply:
Subject under test here is the migration which added these initial statuses in the database which is not what we are interested in.

Reply:
Is the way I've used `TestExecutionStatus.filter` in the permissions test case the right way to do it then? I have used the `filter()` method to filter out all the negative weights.

Reply:
Yes. You've also had this in your previous implementation.